### PR TITLE
Improve developer productivity

### PR DESCRIPTION
**Pull request by nguerrera**
@dsplaisted @dotnet/dotnet-cli

* Build straight to runnable layout. Allows proper edit, test, debug iteration in the IDE without manual command line steps in between.

* Use pack of tasks csproj instead of separate nuspec and project

* Delete unused VsixV3 project

* Restore using sln instead of one project at a time

* Redirect test command output to xunit log and prevent any test command output from showing yellow/red in actual build output

* Eliminate all build warnings and treat future warnings as errors to prevent regression

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj [hunk 1]
```diff
@@ -193,12 +43,44 @@
...
+          AfterTargets="Build"
+          Inputs="@(LayoutFile)"
+          Outputs="@(LayoutFile->'$(PackageLayoutOutputPath)%(TargetPath)')">
+    <Copy SourceFiles="@(LayoutFile)"
+          DestinationFiles="@(LayoutFile->'$(PackageLayoutOutputPath)%(TargetPath)')">
+      <Output TaskParameter="DestinationFiles" ItemName="FileWrites" />
+    </Copy>
+  </Target>
+
+  <Target Name="PackLayout" BeforeTargets="$(GenerateNuspecDependsOn)">
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj]: Should this depend on `CopyAdditionalFilesToLayout`?

**Review by dsplaisted — APPROVED** [review]
💯

Have you inspected the generated NuGet package and compared it to a previous version to make sure it has the right contents?

File: src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj (hunk 1 above)
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj]: Can't hurt

File: test/Microsoft.NET.TestFramework/CommandExtensions.cs [hunk 2]
```diff
@@ -25,5 +25,7 @@ public static ICommand EnsureExecutable(this ICommand command)
             }
             return command;
         }
+
```
Comment (dsplaisted) [inline test/Microsoft.NET.TestFramework/CommandExtensions.cs]: Extra whitespace

File: test/Microsoft.NET.TestFramework/Assertions/CommandResultExtensions.cs [hunk 3]
```diff
@@ -11,5 +12,25 @@ public static CommandResultAssertions Should(this CommandResult commandResult)
         {
             return new CommandResultAssertions(commandResult);
         }
+
+        public static CommandResult AndLog(this CommandResult commandResult, ITestOutputHelper log)
```
Comment (dsplaisted) [inline test/Microsoft.NET.TestFramework/Assertions/CommandResultExtensions.cs]: This looks like dead code.

**Review by dsplaisted — APPROVED** [review]

**Comment by nguerrera** [comment]
@dsplaisted @eerhardt Test failure on Windows is repros locally seems to be due to stage0 CLI upgrade.

I don't know all the workings of how these tool apps are activated, but it seems to think that the tool is standalone and it's looking for hostpolicy in the location of the deps file path.:

```
d:\Src\sdk\.dotnet_cli\dotnet.exe exec --depsfile D:\Src\sdk\packages\.tools\testtool\1.0.0\netcoreapp2.0\TestTool.deps.json --additionalprobingpath D:\Src\sdk\packages D:\Src\sdk\packages\testtool\1.0.0\lib/netcoreapp2.0/TestTool.dll
A fatal error was encountered. The library 'hostpolicy.dll' required to execute the application was not found in 'D:\Src\sdk\packages\.tools\testtool\1.0.0\netcoreapp2.0\'.
```

**Comment by nguerrera** [comment]
@eerhardt helped me sort it out. This is https://github.com/dotnet/core-setup/issues/2376. We can workaround it by normalizing the path, which I'll do.

**Comment by eerhardt** [comment]
See https://github.com/dotnet/core-setup/issues/2376

**Comment by nguerrera** [comment]
@MattGertz FYI for infrastructure-only change to accelerate developer workflow

**Pull request merged**